core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -680,20 +680,20 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   }
 
   private[log] def activeProducersWithLastSequence: Map[Long, Int] = lock synchronized {
-    producerStateManager.activeProducers.map { case (producerId, producerIdEntry) =>
-      (producerId, producerIdEntry.lastSeq)
+    producerStateManager.activeProducers.asScala.map { case (producerId, producerIdEntry) =>
+      (producerId.toLong, producerIdEntry.lastSeq)
     }
-  }
+  }.toMap
```

Comment:
I realized, but since we are creating a new map in this method, I don't see why an immutable Map is needed. Can we just change the result type to be `collection.Map`?

Reply:
Resolving as I added more specific comments in the last review.